src/textual_dev/previews/widgets.py

```diff
@@ -0,0 +1,101 @@
+from textual.app import App, ComposeResult
+from textual.containers import Vertical
+from textual.widgets import Button, ContentSwitcher
+
+from textual_dev.previews.example_widgets import (
+    option_list_example,
+    button_example,
+    checkbox_example,
+    list_item_example,
+    data_table_example,
+    markdown_viewer_example,
+    placeholder_example,
+    pretty_example,
+    directory_tree_example,
+    footer_example,
+    header_example,
+    input_example,
+    label_example,
+    loading_example,
+    markdown_example,
+    radio_button_example,
+    radio_set_example,
+    select_example,
+    selection_list_example,
+    sparkline_example,
+    static_example,
+    switch_example,
+    tree_example,
+)
+
+WIDGETS = {
+    "Button": button_example,
+    "Checkbox": checkbox_example,
+    # ContentSwitcher missing
+    "DataTable": data_table_example,
+    "DirectoryTree": directory_tree_example,
+    "Footer": footer_example,
+    "Header": header_example,
+    "Input": input_example,
+    "Label": label_example,
+    "ListItem": list_item_example,
+    # ListView missing
+    "Loading": loading_example,
+    "MarkdownViewer": markdown_viewer_example,
+    "Markdown": markdown_example,
+    "OptionList": option_list_example,
+    "Placeholder": placeholder_example,
+    "Pretty": pretty_example,
+    # ProgressBar missing
+    "RadioButton": radio_button_example,
+    "RadioSet": radio_set_example,
+    "Select": select_example,
+    "SelectionList": selection_list_example,
+    "Sparkline": sparkline_example,
+    "Static": static_example,
+    "Switch": switch_example,
+    # TabbedContatn missing
+    # Tabs missing
+    # TextLog missing
+    "Tree": tree_example,
+}
+
+
+class WidgetButtons(Vertical):
+    DEFAULT_CSS = """
+    WidgetButtons {
+        dock: left;
+        width: 32;
+        overflow-y: scroll;
+    }
+
+    WidgetButtons > Button {
+        width: 100%;
+    }
+    """
+
+    def compose(self) -> ComposeResult:
+        for widget in WIDGETS.keys():
+            yield Button(widget, id=widget)
```

Comment:
You could take these two lines of code and write them directly inside `WidgetsApp.compose`, instead of `yield WidgetButtons()`.
This would also mean you have to adjust the CSS and put it in `widgets.css`.